content/guides/use-case/wiremock.md

```diff
@@ -0,0 +1,272 @@
+
+    ```javascript
+    require("dotenv").config();
+
+    const express = require("express");
+    const axios = require("axios");
+
+    const router = express.Router();
+    const API_ENDPOINT_BASE = process.env.API_ENDPOINT_BASE;
+    const API_KEY = process.env.ACCUWEATHER_API_KEY;
+
+    console.log('API_ENDPOINT_BASE:', API_ENDPOINT_BASE);  // Log after it's defined
+    console.log('ACCUWEATHER_API_KEY:', API_KEY);          // Log after it's defined
+
+    if (!API_ENDPOINT_BASE) {
+      throw new Error("API_ENDPOINT_BASE is not defined in environment variables");
+    }
+
+    // Only check for API key if not using WireMock
+    if (API_ENDPOINT_BASE !== 'http://localhost:8080' && !API_KEY) {
+      throw new Error("ACCUWEATHER_API_KEY is not defined in environment variables");
+    } 
+    // Function to fetch the location key for the city
+    async function fetchLocationKey(townName) {
+      const { data: locationData } = await 
+    axios.get(`${API_ENDPOINT_BASE}/locations/v1/cities/search`, {
+        params: { q: townName, details: false, apikey: API_KEY },
+      });
+      return locationData[0]?.Key;
+    }
+    ```
+    
+    Keep this terminal window open. 
+
+4. Test the Mocked API
+
+   Open a new terminal window and run the following command to test the mocked API:
+
+   ```console
+   $ curl "http://localhost:5000/api/v1/getWeather?city=Bengaluru"
+   ```
+
+   You should see the following output:
+
+   ```plaintext
+   {"city":"Bengaluru","temperature":27.1,"conditions":"Mostly cloudy","forecasts":[{"date":"2024-09-02T07:00:00+05:30","temperature":83,"conditions":"Partly sunny w/ t-storms"},{"date":"2024-09-03T07:00:00+05:30","temperature":83,"conditions":"Thunderstorms"},{"date":"2024-09-04T07:00:00+05:30","temperature":83,"conditions":"Intermittent clouds"},{"date":"2024-09-05T07:00:00+05:30","temperature":82,"conditions":"Dreary"},{"date":"2024-09-06T07:00:00+05:30","temperature":82,"conditions":"Dreary"}]}%
+   ```
+
+   This indicates that your Node.js application is now successfully routing requests to the WireMock container and receiving the mocked responses
+
+   You might have noticed that you’re trying to use `http://localhost:5000` as the URL instead of port `8080`. This is because your Node.js application is running on port `5000`, and it's routing requests to the WireMock container that's listening on port `8080`.
+
+   > [!TIP]
+   > Before you proceed to the next step, ensure that you stop the node application service.
+
+## Set up a non-containerized Node app to fetch data from the external AccuWeather API 
+
+   To enhance your Node.js application with real-time weather data, you can seamlessly integrate the AccuWeather API. This section of the guide will walk you through the steps involved in setting up a non-containerized Node.js application and fetching weather information directly from the AccuWeather API.
```

Comment:
This almost seems like it's a logical progression of the application, whereas I would suggest framing it as two different modes of operation:

- Use mock data in local development
- Use a live API in production

I think what we might want to show is how you can configure your project to select the appropriate mode for what you're doing. For example, using npm scripts:

- `npm run dev` which runs the app in `NODE_ENV=development`, and uses the mock API via compose
- `npm run start` which runs `NODE_ENV=production` and thereby the production config of the app, toward the live API

Maybe there is a more idiomatic way to do it, but you get my point.

Reply:
Hello @dvdksn I tried modifying the initial para with the following:

``` 
In this guide, you'll learn how to:

- Use Docker to launch up a WireMock container.
- Use mock data in the local development without relying on an external API
- Use a Live API in production to fetch real-time weather data from AccuWeather.
```

Reply:
I was going through your 2nd point where you suggested using appropriate modes. 
As we're focusing on container-supported development workflow and not using Compose for the apps, I am wondering how to pass the NODE_ENV=development in that case.

Reply:
in `package.json`:

```
  "scripts": {
    "start": "NODE_ENV=production node src/index.js",
    "dev": "NODE_ENV=development node src/index.js"
  },
```

Reply:
@dvdksn LMK if there is any action item from my end. Does it look good?

Reply:
I updated https://github.com/dockersamples/wiremock-node-docker/blob/main/accuweather-api/package.json with the entries.